content_restrictions/content_restrictions.py

```diff
@@ -81,12 +86,54 @@ class XblockContentRestrictions(
         default="",
     )
 
+    ip_restriction = Boolean(
+        display_name=_("IP Restriction"),
+        help=_(
+            "When enabled, the content will be restricted by a range of IP addresses."
+        ),
+        scope=Scope.settings,
+        default=False,
+    )
+
+    ip_whitelist = List(
+        display_name=_("IP Whitelist"),
+        help=_(
+            "List of IP addresses or IP address ranges from which the content can be accessed."
+            " The compatible protocols are IPv4 and IPv6. If the learner's IP address is not in the"
+            " whitelist, they will not have access to the content. The range should be in the"
+            ' CIDR format, e.g. (IPv4) "192.168.1.0/24", in this case only IP addresses from '
+            ' "192.168.1.0" to "192.168.1.255" can access to the content, or (IPv6) "2001:db8:85a3::/48",'
+            ' in this case only IP addresses from "2001:db8:85a3::" to "2001:db8:85a3:ffff:ffff:ffff:ffff:ffff"'
+            ' can access to the content. Also, a single IP address can be provided, e.g. (IPv4) "172.16.0.0",'
+            ' or (IPv6) "7ac:264a:dd69::". Alternatively, is possible mix IP addresses and IP ranges'
+            ' with different formats, e.g. ["192.168.1.0/24", "172.16.0.0", "65c1:e700::/24", "5c87::"]'
+        ),
+        scope=Scope.settings,
+        default=[],
```

Comment:
Why should the user include the brackets? Could we handle this as a comma-separated string instead? What would be less error-prone?

Reply:
I tested it both ways, and I found it more understandable with the list field.

By default, the square brackets are already added, and the user should only include the addresses or ranges as list items, as is currently done to add an xblock in the advanced module list.

Especially when there are many ips or ranges I think it is more confusing to handle it with a string separated by commas, it is difficult to differentiate where one or the other begins, since there can be characters from `,` `.`  `:` practically in a row, e.g:
```
10.16.0.1,2000:0db8:85a3:0000:0000:8a2e:0370:7334,2400:cb00::/32,f380::1

or

["10.16.0.1", "2000:0db8:85a3:0000:0000:8a2e:0370:7334", "2400:cb00::/32", "f380::1"]
```

Maybe we could explain to the user a little better how to add the IPs, so it would be clearer, what do you think?

Reply:
Yes, that renders better than the str. What would happen if I accidentally removed the list and then saved it? Could we test that?

Reply:
If I delete the list and save, the field is stored as `None`. To avoid errors, we can check before iterating the list, if the field is `None`, in that case, we set it as an empty list (`[]`).

Also in the field definition we add `enforce_type=True`, to avoid setting an invalid value such as a number or a string.

Done: https://github.com/eduNEXT/xblock-content-restrictions/commit/b06b14b5608563e7720d582f9aa2280d9372f071